From f581bbdbceee04fd1925acde16640b7f26e5a48b Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 9 Jul 2017 08:23:28 -0400
Subject: [PATCH] Memory leak in mat coder

A special crafted file create a memory leak in MAT coder

bug: https://github.com/ImageMagick/ImageMagick/issues/553
bug-debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=870015
origin: https://github.com/ImageMagick/ImageMagick/commit/8985ed08f01d465ee65ab5a106186b3868b6f601

(cherry picked from commit 8985ed08f01d465ee65ab5a106186b3868b6f601)
---
 coders/mat.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/coders/mat.c b/coders/mat.c
index d160c314e..df7751a04 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -932,7 +932,11 @@ static Image *ReadMATImage(const ImageInfo *image_info,ExceptionInfo *exception)
     goto MATLAB_KO;    /* unsupported endian */
 
   if (strncmp(MATLAB_HDR.identific, "MATLAB", 6))
-MATLAB_KO: ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+    {
+MATLAB_KO:
+      clone_info=DestroyImageInfo(clone_info);
+      ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+    }
 
   filepos = TellBlob(image);
   while(!EOFBlob(image)) /* object parser loop */
